src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,159 @@
+            int newTableIndex;
+            while (transferredNode != null) {
+                newTableIndex = calculateIndex(transferredNode.key);
+                // detached function
+                if (newTable[newTableIndex] != null) {
+                    putTransferredNodeToExtendedTableNode(newTableIndex, newTable, transferredNode);
+                } else {
+                    newTable[newTableIndex] = new Node<>(transferredNode.key,
+                            transferredNode.value, null);
+                }
+                transferredNode = transferredNode.next;
+            }
+        }
+        return newTable;
+    }
+
+    private void putTransferredNodeToExtendedTableNode(
+            int newTableIndex, Node<K,V>[] newTable, Node<K,V> transferredNode) {
+        Node<K,V> currentNodeOfTableToBeAddedTo = newTable[newTableIndex];
+        boolean areNodeKeysEqual = false;
+        while (currentNodeOfTableToBeAddedTo.next != null) {
+            if (transferredNode.key == currentNodeOfTableToBeAddedTo.key
+                    || transferredNode.key != null
+                    && transferredNode.key.equals(currentNodeOfTableToBeAddedTo.key)) {
+                currentNodeOfTableToBeAddedTo.value = transferredNode.value;
+                areNodeKeysEqual = true;
+                break;
+            } else {
+                currentNodeOfTableToBeAddedTo = currentNodeOfTableToBeAddedTo.next;
+            }
+        }
+        if (!areNodeKeysEqual) {
+            currentNodeOfTableToBeAddedTo.next =
+                    new Node<K, V>(transferredNode.key, transferredNode.value, null);
+        }
+    }
+
+    private boolean putNewNodeToCurrentCapacityTableNode(
+            Node<K,V>[] tableToBeAddedTo, int index, Node<K,V> nodeToPut,
+            boolean isSizeIncrease) {
+        Node<K,V> currentNodeOfTableToBeAddedTo = tableToBeAddedTo[index];
+        while (currentNodeOfTableToBeAddedTo != null) {
+            if (nodeToPut.key == currentNodeOfTableToBeAddedTo.key
+                    || nodeToPut.key != null
+                    && nodeToPut.key.equals(currentNodeOfTableToBeAddedTo.key)) {
+                currentNodeOfTableToBeAddedTo.value = nodeToPut.value;
+                isSizeIncrease = false;
+                break;
+            }
+            if (currentNodeOfTableToBeAddedTo.next == null) {
+                break;
+            }
+            currentNodeOfTableToBeAddedTo = currentNodeOfTableToBeAddedTo.next;
+        }
+        if (isSizeIncrease) {
+            currentNodeOfTableToBeAddedTo.next =
+                    new Node<K, V>(nodeToPut.key, nodeToPut.value, null);
+        }
+        return isSizeIncrease;
+    }
+
+    private void increaseCurrentCapacityAndThreshold() {
+        currentCapacity = currentCapacity << 1;
+        threshold = threshold << 1;
+    }
+
+    private static class Node<K,V> {
+        private K key;
+        private V value;
+        private Node<K,V> next;
```

Comment:
The `hash` field in the `Node` class is declared but never used. Consider removing it to clean up the code and avoid confusion, as it does not contribute to the current implementation.